zcash_primitives/src/zip32.rs

```diff
@@ -159,6 +160,18 @@ impl DiversifierKey {
         Self::try_diversifier_internal(&ff, j)
     }
 
+    /// Decrypts a diversifier using this diversifier key to obtain the
+    /// diversifier index from which it was originally created.
+    pub fn diversifier_index(&self, d: &Diversifier) -> Option<DiversifierIndex> {
+        let ff = FF1::<Aes256>::new(&self.0, 2).unwrap();
+        let dec = ff
+            .decrypt(&[], &BinaryNumeralString::from_bytes_le(&d.0[..]))
+            .ok()?;
```

Comment:
We can unwrap here; `FF1::decrypt` only returns an error if the numeral string is in the incorrect radix, but we hard-code the matching radixes here (`2` and `BinaryNumeralString`):
```suggestion
            .unwrap();
```

Reply:
Opened https://github.com/str4d/fpe/issues/24 to clean up the API now that const generics are a thing.